docs/Pool.md

```diff
@@ -0,0 +1,107 @@
+# Class: Pool
```

Comment:
Should this be linked from the README?

Reply:
Yes, once I finish the Agent file I'm going to rip everything out of the current readme and replace it with links to these documents (+ include some other things I've been thinking about for the main readme file)